v3/onesignal-metabox/onesignal-metabox.js

```diff
@@ -32,4 +32,38 @@ window.addEventListener("DOMContentLoaded", () => {
     setDisplay(customiseWrap, customisePost.checked);
     setDisabled(customiseWrapChild, !customisePost.checked);
   });
+
+  // make sure WordPress editor and API are available
+  if (typeof wp === 'undefined' || !wp.data || !wp.data.select) {
+    console.warn('wp.data is not available.');
+    return;
+  }
+
+  const editorStore = wp.data.select('core/editor');
+
+  // track initial state of checkbox
+  const osUpdateCheckbox = document.querySelector('#os_update');
+  const wasCheckedInitially = osUpdateCheckbox ? osUpdateCheckbox.checked : false;
+
+  // track previous post status to detect changes
+  let previousStatus = editorStore.getCurrentPostAttribute('status');
+
+  // subscribe to state changes
+  wp.data.subscribe(() => {
+    const currentStatus = editorStore.getCurrentPostAttribute('status');
+
+    // check if the post status changed to "publish"
+    if (previousStatus !== currentStatus && currentStatus === 'publish') {
+        previousStatus = currentStatus;
+
+        if (wasCheckedInitially) {
+            // uncheck the os_update checkbox
+            if (osUpdateCheckbox && osUpdateCheckbox.checked) {
+                osUpdateCheckbox.checked = false;
+            }
+        }
```

Comment:
Think we can just reuse the element defined on [L2](https://github.com/OneSignal/OneSignal-WordPress-Plugin/pull/342/files#diff-097caacc8da384c407236119e259d18de27ec1fec7abe7393bc880133bfe60c9R2)
```suggestion
  const wasCheckedInitially = sendPost ? sendPost.checked : false;

  // track previous post status to detect changes
  let previousStatus = editorStore.getCurrentPostAttribute('status');

  // subscribe to state changes
  wp.data.subscribe(() => {
    const currentStatus = editorStore.getCurrentPostAttribute('status');

    // check if the post status changed to "publish"
    if (previousStatus !== currentStatus && currentStatus === 'publish') {
        previousStatus = currentStatus;

        if (wasCheckedInitially) {
            // uncheck the sendPost checkbox
            if (sendPost && sendPost.checked) {
                sendPost.checked = false;
            }
        }
```

Reply:
Good catch. Added a fix for that